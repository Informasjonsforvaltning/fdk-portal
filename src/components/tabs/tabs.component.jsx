import React from 'react';
import PropTypes from 'prop-types';
import { TabContent, TabPane, Nav, NavItem, NavLink } from 'reactstrap';
import classnames from 'classnames';
import './tabs.component.scss';

const renderTabContent = (tabContent, activeTab) => {
  if (!tabContent) {
    return null;
  }

  const items = items =>
    items.map((item, index) => (
      <TabPane key={index} tabId={index}>
        {item.body}
      </TabPane>
    ));

  return (
    <TabContent className="pb-4" activeTab={activeTab}>
      {items(tabContent)}
    </TabContent>
  );
};

export class Tabs extends React.Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.renderNavItems = this.renderNavItems.bind(this);
    this.state = {
      activeTab: 0
    };
  }

  toggle(tab) {
    const { activeTab } = this.state;
    if (activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  }

  renderNavItems(tabContent, activeTab) {
    if (!tabContent) {
      return null;
    }

    const navItems = (items, activeTab) =>
      items.map((item, index) => (
        <NavItem key={index}>
          <NavLink
            className={classnames({ active: activeTab === index })}
            onClick={() => {
              this.toggle(index);
            }}
          >
            {item.title}
          </NavLink>
        </NavItem>
      ));

    return <Nav tabs>{navItems(tabContent, activeTab)}</Nav>;
  }

  render() {
    const { activeTab } = this.state;
    const { tabContent } = this.props;
    return (
      <>
        {this.renderNavItems(tabContent, activeTab)}
        {renderTabContent(tabContent, activeTab)}
      </>
    );
  }
}

Tabs.defaultProps = {
  tabContent: null
};

Tabs.propTypes = {
  tabContent: PropTypes.array
};
