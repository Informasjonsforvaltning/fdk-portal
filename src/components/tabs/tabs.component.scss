.nav-tabs {
  .nav-item {
    cursor: pointer;
  }
}
