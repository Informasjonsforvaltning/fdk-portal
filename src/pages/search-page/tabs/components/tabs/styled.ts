import styled from 'styled-components';

import AllIconBase from '../../../../../images/icon-catalog-all-md.svg';
import DatasetIconBase from '../../../../../images/icon-catalog-dataset-md.svg';
import ApiIconBase from '../../../../../images/icon-catalog-api-md.svg';
import ConceptIconBase from '../../../../../images/icon-catalog-concept-md.svg';
import InfomodIconBase from '../../../../../images/icon-catalog-infomod-md.svg';

const Tabs = styled.ul`
  align-items: center;
  display: flex;
  justify-content: center;
  flex-wrap: nowrap;
  margin-bottom: 0;
  padding-left: 0;
  @media (max-width: 768px) {
    justify-content: space-around;
  }
`;

const Label = styled.span`
  margin-left: 0.5em;
  white-space: nowrap;
  @media (max-width: 768px) {
    display: none;
    margin-left: 0;
  }
`;

const AllIcon = styled(AllIconBase)`
  height: 1.5em;
  @media (min-width: 768px) {
    height: 1em;
  }
`;

const DatasetIcon = styled(DatasetIconBase)`
  border-radius: 50%;
  padding: 0.5em;
  background-color: ${({ theme }) => theme.colors.dataset.light};
  height: 3em;
  fill: ${({ theme }) => theme.colors.dataset.dark};

  @media (min-width: 768px) {
    height: 2em;
  }
`;

const ApiIcon = styled(ApiIconBase)`
  border-radius: 50%;
  padding: 0.5em;
  background-color: ${({ theme }) => theme.colors.api.light};
  height: 3em;
  fill: ${({ theme }) => theme.colors.api.dark};

  @media (min-width: 768px) {
    height: 2em;
  }
`;

const ConceptIcon = styled(ConceptIconBase)`
  border-radius: 50%;
  padding: 0.5em;
  background-color: ${({ theme }) => theme.colors.concept.light};
  height: 3em;
  fill: ${({ theme }) => theme.colors.concept.dark};

  @media (min-width: 768px) {
    height: 2em;
  }
`;

const InfomodIcon = styled(InfomodIconBase)`
  border-radius: 50%;
  padding: 0.5em;
  background-color: ${({ theme }) => theme.colors.infomod.light};
  height: 3em;
  fill: ${({ theme }) => theme.colors.infomod.dark};

  @media (min-width: 768px) {
    height: 2em;
  }
`;

export default {
  Tabs,
  Label,
  AllIcon,
  DatasetIcon,
  ApiIcon,
  ConceptIcon,
  InfomodIcon
};
