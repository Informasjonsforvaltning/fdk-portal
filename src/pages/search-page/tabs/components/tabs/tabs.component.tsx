import React, { FC, memo } from 'react';
import { RouteComponentProps, withRouter } from 'react-router';

import SC from './styled';
import Tab from '../tab/tab.component';
import localization from '../../../../../lib/localization';
import {
  PATHNAME_APIS,
  PATHNAME_CONCEPTS,
  PATHNAME_DATASETS,
  PATHNAME_INFORMATIONMODELS
} from '../../../../../constants/constants';
import { getLinkForTab } from '../../../search-location-helper';

interface Props {
  countDatasets: number;
  countConcepts: number;
  countApis: number;
  countInformationModels: number;
}

const Tabs: FC<Props & RouteComponentProps> = ({
  countDatasets,
  countApis,
  countConcepts,
  countInformationModels,
  location
}) => {
  const renderLabel = () => {
    switch (location.pathname) {
      case PATHNAME_DATASETS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countDatasets}`,
          localization.datasetLabel.toLowerCase()
        );
      }
      case PATHNAME_APIS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countApis}`,
          localization.apiLabel.toLowerCase()
        );
      }
      case PATHNAME_CONCEPTS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countConcepts}`,
          localization.conceptLabel.toLowerCase()
        );
      }
      case PATHNAME_INFORMATIONMODELS: {
        return localization.formatString(
          localization.showHitsFor,
          `${countInformationModels}`,
          localization.informationModelLabel.toLowerCase()
        );
      }
      default: {
        return null;
      }
    }
  };
  return (
    <>
      <SC.Tabs>
        <Tab
          active={location.pathname === PATHNAME_DATASETS}
          tabLink={getLinkForTab(location, PATHNAME_DATASETS)}
          label={localization.datasetLabel}
        >
          <SC.DatasetIcon />
          <SC.Label>
            {localization.page.datasetTab}&nbsp;({countDatasets})
          </SC.Label>
        </Tab>

        <Tab
          active={location.pathname === PATHNAME_APIS}
          tabLink={getLinkForTab(location, PATHNAME_APIS)}
          label={localization.apiLabel}
        >
          <SC.ApiIcon />
          <SC.Label>
            {localization.page.apiTab}&nbsp;({countApis})
          </SC.Label>
        </Tab>

        <Tab
          active={location.pathname === PATHNAME_CONCEPTS}
          tabLink={getLinkForTab(location, PATHNAME_CONCEPTS)}
          label={localization.conceptLabel}
        >
          <SC.ConceptIcon />
          <SC.Label>
            {localization.page.termTab}&nbsp;({countConcepts})
          </SC.Label>
        </Tab>

        <Tab
          active={location.pathname === PATHNAME_INFORMATIONMODELS}
          tabLink={getLinkForTab(location, PATHNAME_INFORMATIONMODELS)}
          label={localization.informationModelLabel}
        >
          <SC.InfomodIcon />
          <SC.Label>
            {localization.page.informationModelTab}&nbsp;(
            {countInformationModels})
          </SC.Label>
        </Tab>
      </SC.Tabs>
      <SC.MobileLabel>{renderLabel()}</SC.MobileLabel>
    </>
  );
};

export default withRouter(memo(Tabs));
