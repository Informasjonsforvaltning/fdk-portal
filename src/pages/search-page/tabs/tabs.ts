/* istanbul ignore file */
import TabSC from './components/tab/styled';

export { default as Tabs } from './components/tabs/tabs.component';
export { default as Tab } from './components/tab/tab.component';

export const SC = {
  Tab: TabSC
};
