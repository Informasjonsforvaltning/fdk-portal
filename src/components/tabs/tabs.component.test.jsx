import React from 'react';
import { shallow } from 'enzyme';
import { Tabs } from './tabs.component';

test('should render Tabs correctly with no props', () => {
  const minWrapper = shallow(<Tabs />);
  expect(minWrapper).toMatchSnapshot();
});

test('should render Tabs correctly', () => {
  const defaultProps = {
    tabContent: [
      { title: 'Tab content 1', data: 'Data tab content 1' },
      { title: 'Tab content 2', data: 'Data tab content 2' }
    ]
  };
  const wrapper = shallow(<Tabs {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
