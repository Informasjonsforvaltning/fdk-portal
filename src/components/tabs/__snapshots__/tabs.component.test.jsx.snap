// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Tabs correctly 1`] = `
<Fragment>
  <Nav
    tabs={true}
    tag="ul"
    vertical={false}
  >
    <NavItem
      key="0"
      tag="li"
    >
      <NavLink
        className="active"
        onClick={[Function]}
        tag="a"
      >
        Tab content 1
      </NavLink>
    </NavItem>
    <NavItem
      key="1"
      tag="li"
    >
      <NavLink
        className=""
        onClick={[Function]}
        tag="a"
      >
        Tab content 2
      </NavLink>
    </NavItem>
  </Nav>
  <TabContent
    activeTab={0}
    className="pb-4"
    tag="div"
  >
    <TabPane
      key="0"
      tabId={0}
      tag="div"
    />
    <TabPane
      key="1"
      tabId={1}
      tag="div"
    />
  </TabContent>
</Fragment>
`;

exports[`should render Tabs correctly with no props 1`] = `<Fragment />`;
